corehq/apps/reports/templates/reports/form/partials/single_form_tree.html

```diff
@@ -53,9 +53,15 @@
         <td class="span6" title="{{ question.get_formatted_response }}">
             {% if question.options %}
                 <div class="form-data-readable">
-                {% for option in question.options %}
-                    {% if option.value in question.response.split %}
-                        {{ option.label|linebreaksbr }}
+                {% for response in question.response.split %}
+                    {% if response in question.option_values %}
+                        {% for option in question.options %}
+                            {% if option.value == response %}
```

Comment:
Thats quite a puzzle.. :)
nit: if this can be made simpler/easier to understand, it would be great.

Reply:
will add a comment - I can't think how to make the code simpler but agree logic in templates gets hard to read quickly

Reply:
...having said that, I took another look and cleaned up the code a tiny bit :)

Reply:
Aha! great... thanks for the clean up..